python/graphscope/client/session.py

```diff
@@ -913,20 +918,36 @@ def gremlin(self, graph, engine_params=None):
             graph.vineyard_id in self._interactive_instance_dict
             and self._interactive_instance_dict[graph.vineyard_id] is not None
         ):
-            return self._interactive_instance_dict[graph.vineyard_id]
+            interactive_query = self._interactive_instance_dict[graph.vineyard_id]
+            if interactive_query.status == "running":
+                return interactive_query
+            elif interactive_query.status == "pending":
+                begin_time = time.time()
+                while True:
+                    time.sleep(1)
+                    if interactive_query.status == "running":
+                        return interactive_query
+                    if time.time() - begin_time >= timeout_seconds:
```

Comment:
After timeout, is the creating thread still alive? There can be chances that the starting procedure actually takes 70s, for example. What will happen if this method already raised timeout, and after sometime the thread successfully launched the server?

Reply:
The `timeout` here is a little bit confused. `while True` is ok, as the status is either running or failed eventually after timeout.
```python
while True:
    time.sleep(1)
    if interactive_query.status == InteractiveQueryStatus.Running
        return interactive_query
    elif interactive_query.status == InteractiveQueryStatus.Failed:
        raise InteractiveEngineInternalError(interactive_query.error_msg)
```